packages/block-editor/src/components/use-block-drop-zone/index.js

```diff
@@ -422,6 +422,10 @@ export default function useBlockDropZone( {
 				const [ targetIndex, operation, nearestSide ] =
 					dropTargetPosition;
 
+				if ( isZoomOutMode() && operation !== 'insert' ) {
+					return;
+				}
```

Comment:
There are too many of these conditions creeping in which disallow non-"insert" events in Zoom Out mode. Whilst this is desired behaviour I'd like to codify the concept and have it made reusable. One for a follow up.

Reply:
For now would it work to move this check up to the existing `isZoomOutMode()` check earlier in this code block so that the zoomed out mode checks are consolidated?

Leaving to follow-ups is also a valid approach! My (very) weakly held opinion here is that sometimes it can be tricky to change some of these rules after the fact if we leave them for a while as the drag and drop logic can be a bit challenging to reason about — I know I struggle looking at some of my past code changes to this area of the codebase 😅

Reply:
I'd love to do that but the issue is that `operation` isn't available until this point. That in turn is dependent on other steps above. We could bring all the ZoomOut checks down to _this_ location, but then we'd be doing unnecessary work rather than existing early.

Reply:
I gave this a go but moving into the consumer causes the fix to break when testing in browser. I can't narrow down why that is.

Reply:
Ok I ended up moving the whole thing to the `dragEnd` handler. That has the desired effect.